### PR TITLE
build: Bump symfony/* to 6.4

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] https://github.com/nextcloud/3rdparty/pull/1956

## Summary

| Prod Packages                      | Operation | Base    | Target  |
|------------------------------------|-----------|---------|---------|
| psr/log                            | Upgraded  | 2.0.0   | 3.0.2   |
| stecman/symfony-console-completion | Upgraded  | 0.11.0  | v0.13.0 |
| symfony/console                    | Upgraded  | v5.4.35 | v6.4.12 |
| symfony/event-dispatcher           | Upgraded  | v5.4.26 | v6.4.8  |
| symfony/event-dispatcher-contracts | Upgraded  | v3.0.2  | v3.5.0  |
| symfony/http-foundation            | Upgraded  | v5.4.25 | v6.4.12 |
| symfony/mailer                     | Upgraded  | v5.4.22 | v6.4.12 |
| symfony/mime                       | Upgraded  | v5.4.19 | v6.4.12 |
| symfony/polyfill-intl-idn          | Upgraded  | v1.29.0 | v1.31.0 |
| symfony/polyfill-php83             | New       | -       | v1.31.0 |
| symfony/process                    | Upgraded  | v5.4.34 | v6.4.12 |
| symfony/routing                    | Upgraded  | v5.4.25 | v6.4.12 |
| symfony/service-contracts          | Upgraded  | v3.0.2  | v3.5.0  |
| symfony/polyfill-php72             | Removed   | v1.29.0 | -       |
| symfony/polyfill-php73             | Removed   | v1.28.0 | -       |

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by kesselb — APPROVED** [review]

File: build/psalm-baseline.xml [hunk 1]
```diff
@@ -1285,12 +1286,24 @@
       <code><![CDATA[getFilteredValues]]></code>
     </TooManyArguments>
   </file>
+  <file src="core/Command/Db/ConvertFilecacheBigInt.php">
```
Comment (susnux) [inline build/psalm-baseline.xml:1289]: Just missing a `use Symfony\Component\Console\Helper\QuestionHelper;` ?

File: core/Command/Db/ConvertFilecacheBigInt.php [hunk 2]
```diff
@@ -89,6 +89,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 		$output->writeln('<comment>This can take up to hours, depending on the number of files in your instance!</comment>');
 
 		if ($input->isInteractive()) {
+			/** @var QuestionHelper $helper */
```
Comment (come-nc) [inline core/Command/Db/ConvertFilecacheBigInt.php:92]: Can we document that in Base so that at least method extending Base do not need to type it each time?

File: apps/files_external/lib/Command/Option.php [hunk 3]
```diff
@@ -35,10 +35,10 @@ protected function configure(): void {
 	 */
 	protected function getOption(StorageConfig $mount, $key, OutputInterface $output): void {
 		$value = $mount->getMountOption($key);
-		if (!is_string($value)) { // show bools and objects correctly
+		if (!is_string($value) && json_decode(json_encode($value)) === $value) { // show bools and objects correctly
```
Comment (come-nc) [inline apps/files_external/lib/Command/Option.php:38]: This needs a comment explaining what it does.

Comment (nickvergessen) [reply apps/files_external/lib/Command/Option.php:38]: Copied from `apps/files_external/lib/Command/Config.php` I assumed they should be the same?

File: build/psalm-baseline.xml (hunk 1 above)
Comment (nickvergessen) [reply build/psalm-baseline.xml:1289]: Right, not sure what went wrong there on the import

File: apps/files_external/lib/Command/Option.php [hunk 4]
```diff
@@ -35,10 +35,10 @@ protected function configure(): void {
...
 			$value = json_encode($value);
```
Comment (come-nc) [inline apps/files_external/lib/Command/Option.php:39]:
```suggestion
			$value = json_encode($value) ?: (string)$value;
```
And you can revert the change above and the string cast below.

File: core/Command/Db/ConvertFilecacheBigInt.php (hunk 2 above)
Comment (nickvergessen) [reply core/Command/Db/ConvertFilecacheBigInt.php:92]: Can we document based on the arguments? I don't think so

Comment (come-nc) [reply core/Command/Db/ConvertFilecacheBigInt.php:93]: Ah yeah I see.

**Review by come-nc — APPROVED** [review]

**Comment by nickvergessen** [comment]
> Maybe we should simply add in Base:

I'm up for that, if we add a public interface. I don't feel like typing 3rdparty with implied assumptions

**Comment by nickvergessen** [comment]
/backport e6cd1051dac570826c969c7e7372a38b192056fa to stable29

**Comment by nickvergessen** [comment]
/backport e6cd1051dac570826c969c7e7372a38b192056fa to stable28
